org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ScalaProject.scala

```diff
@@ -67,15 +71,27 @@ object ScalaProject {
 
   /** Listen for [[IWorkbenchPart]] event and takes care of loading/discarding scala compilation units.*/
   private class ProjectPartListener(project: ScalaProject) extends PartAdapter with HasLogger {
+    override def partActivated(part: IWorkbenchPart): Unit = {
```

Comment:
Does this restart the PC on every focus to the editor? So going through different editors will restart the PC each time, if there are errors?